sql/item_strfunc.cc

```diff
@@ -1449,6 +1449,68 @@ String *Item_func_sformat::val_str(String *res)
   return null_value ? NULL : res;
 }
 
+
+#include <openssl/rand.h>
+#include <openssl/err.h>
+
+bool Item_func_random_bytes::fix_length_and_dec() 
+{
+  used_tables_cache|= RAND_TABLE_BIT;
+  max_length= MAX_BLOB_WIDTH;
+  set_maybe_null();
```

Comment:
I don't think that's right. `random_bytes(X)` can only be NULL if the argument X is NULL. This is taken into account automatically, in `Item_func::fix_fields()` (take a look there). You only need to set maybe_null if your function doesn't follow the most common rule "return NULL if any of the arguments is NULL". Like `ISNULL()` which does not return NULL if the argument is NULL. Or `div` which can return NULL (`1 div 0`)  even when both arguments are not NULL.

Reply:
By the way, this is one of the things you would've seen in `show create table`. I'd expect the column for `random_bytes(100)` to be `NOT NULL` and because of `set_maybe_null()` it won't be.

So, I'd suggest to add `show create table` to the test first. And then remove `set_maybe_null()` and see what changes.

Reply:
I've checked several Item_str_func descendants -- they all call set_maybe_null, like the Item_func_repeat mentioned in the comment above.

I suggest to remain consistent -- if we change this behavior, then we change it everywhere